flytekit/models/literals.py

```diff
@@ -794,4 +798,9 @@ def from_flyte_idl(cls, pb2_object):
             scalar=Scalar.from_flyte_idl(pb2_object.scalar) if pb2_object.HasField("scalar") else None,
             collection=collection,
             map=LiteralMap.from_flyte_idl(pb2_object.map) if pb2_object.HasField("map") else None,
+            # TODO: explain that string always have a value set in protobufs, which means
```

Comment:
why is this a todo?  you just want to add more text to the comment?  or document it in flyte cookbook or something?

Reply:
you can explain with this example:
```
def test_jcdskl():
    from flyteidl.core.literals_pb2 import KeyValuePair
    from io import BytesIO

    k1 = KeyValuePair(key="fjdksl", value=None)
    bytIO = BytesIO()
    bytIO.write(k1.SerializeToString())

    print(bytIO.getvalue())
    bytIO.seek(0)

    k2 = KeyValuePair()
    k2.ParseFromString(bytIO.getvalue())
    print(k2)
    assert k2.value == ""
```

Reply:
This was a note to self as I was debating if there was any value in having wrapping the value of hash in another type (as opposed to a string). The original problem was the impossibility of differentiating between `None` and empty string, which is a quirk of how strings work in protobufs. 

We could use https://github.com/envoyproxy/protoc-gen-validate#protoc-gen-validate-pgv to handle the case of empty strings (by setting `min_bytes`), but definitely not in this PR.